src/components/budget/budget.service.ts

```diff
@@ -168,20 +163,16 @@ export class BudgetService {
 
     const result = await this.budgetRepo.readOne(id, session, view);
 
-    const perms = await this.authorizationService.getPermissions({
-      resource: Budget,
-      sessionOrUserId: session,
-      dto: result as ResourceShape<Budget>['prototype'],
-    });
+    const privs = this.privileges.for(session, Budget, result).all;
 
     const securedProps = parseSecuredProperties(
       result,
-      perms as PermissionsOf<Budget>,
+      privs,
       Budget.SecuredProps,
     );
 
     let records = null;
-    if (perms.records.canRead) {
+    if (privs.records.read) {
```

Comment:
FYI this could be
```ts
privileges.can('read, 'records')
```
You don't have to use the `.all` proxy.

Reply:
Yea. I think I learned more as I went and forgot to come back and check these 🤦